### ISSUE TITLE
MissingMethodException -- Microsoft.ReportingServices.Common.DateTimeUtil.TryParseDateTime

### ISSUE THREAD

**Issue by mkamoski** [ilich/MvcReportViewer#81]
Dear Ilich (and all MvcReportViewer users) -- 

Please help.

I am using  Microsoft SQL Server Reporting Services Version 10.50.6000.34 now in a C# MVC app that targets the .NET Framework 4.5 here, and I am using the MvcReportViewer Version 0.6.3 too.

What I have found is that if my report does NOT contain any DateTime parameters in it, then the reports run and display fine.

However, if my reports contain a DateTime parameter, then they break with the following exception...
---------------------------
Message from webpage
---------------------------
errorType='System.MissingMethodException'

error='Method not found: 'Boolean Microsoft.ReportingServices.Common.DateTimeUtil.TryParseDateTime(System.String, System.IFormatProvider, System.DateTimeOffset ByRef, Boolean ByRef)'.'
---------------------------
OK   
---------------------------

So is there any way around this?

Do I need to get the source for MvcReportViewer and use a different coding approach to get the same result?

Do I need to convert my DateTime parameters and define them as String parameters and then handle the casting internally in my report itself?

Or something else?

Any thoughts or suggestions would be helpful.

Please advise.

Thanks.

-- Mark Kamoski

**Comment by ilich** [comment]
It seems to be a compatibility issue between Report Viewer assbemblies in GAC and in the bin/ folder. I think this is a generic issue and you will have it with all other MvcReportViewer versions. 

I found the discussion about similar problem on Stack Overflow.

http://stackoverflow.com/questions/19523249/sql-server-reporting-services-error-running-a-report-azure-web-site-deploymen

**Comment by mkamoski** [comment]
Ilich (and all) -- 

It turns out that the root cause was the dreaded "MissingMethodException Method not found: 'Boolean Microsoft.ReportingServices.Common.DateTimeUtil.TryParseDateTime" for all of my woes, or so it seems to me and I found a simple (but somewhat hack) way of getting past it.

It seems to me that there was some kind of version incompatibility issue, and I do not know exactly what is on the server, and I do not have control of what is on the server.

But the MissingMethodException issue was only expressed on reports that had DateTime parameters, hence the underlying call to TryParseDateTime. So, all I did was convert my parameters with data-type DateTime to be data-type text. I know it is a workaround and kind of a hack, but I used it anyway and it worked beautifully. The downside is that I do not get the nice popup calendar inputs on the report toolbar when it renders, but that is OK. Sure, if they put a non-date value in the textbox then the report will say something like "cannot convert input to valid date time" but it would have said that anyway if the user put in junk.

So problem solved, all is well, and thanks so much for all your help.

Thanks.

-- Mark Kamoski
